src/ui.js

```diff
@@ -100,20 +104,29 @@ function handleFileSelect(map, evt) {
         modal.addSuccess();
     };
 
+    const handleTrack = async (file, name) => {
+        for (const track of await parseTrack(file, name)) {
+            track.filename = name;
+            tracks.push(track);
+            map.addTrack(track);
+            modal.addSuccess();
+        }
+    };
+
     const handleFile = async file => {
-        const extension = file.name.split('.').pop().toLowerCase();
+        let name = file.name;
         try {
-            switch (extension) {
-                case 'gpx':
-                    return await handleGPX(file);
-                case 'jpg':
-                case 'jpeg':
-                    return await handleImage(file);
-                default:
-                    throw 'Unsupported file format';
+            if (/\.jpe?g$/i.test(name)) {
+                return await handleImage(file);
+            }
+            const contents = /\.gz$/i.test(name) ? await readGz(file) : await readFile(file);
+            name = name.replace(/\.gz$/i, '');
+            if (/\.(gpx|tcx)$/i.test(name)) {
```

Comment:
Is it worth storing whether the file is a TCX or a GPX file, or is it enough to look at what the root element within it is ([src/track.js:21](https://github.com/erik/derive/pull/18/files#diff-0d92326ea3a7159fcb03bae37efad9b3R71))?